### ISSUE TITLE
Improve the entitlements debug/release experience

### ISSUE THREAD

**Issue by Redth** [xamarin/xamarin-macios#18344]
Looking at recent changes to our MAUI templates:

https://github.com/dotnet/maui/blob/main/src/Templates/src/templates/maui-mobile/MauiApp.1.csproj#L64-L73

This is a bit verbose and would be preferable to not include in the templates by default.  

How can we improve the implicit behaviour here?

### Entitlements.Debug.plist
- com.apple.security.get-task-allow
 
I'm not sure exactly why this one was added... perhaps for debugging the webview, or maybe for secure storage API's in debug mode

### Entitlements.Release.plist
- com.apple.security.app-sandbox
- com.apple.security.network.client

These are required for publishing catalyst apps to the store.

## Proposal

1. Seems like a good time to introduce an item group of something like `PartialEntitlementsPlist` that allows multiple entitlements files to be added to the build to be merged together.
2. Is there a good reason _not_ to implicitly merge the release plist items into a release maccatalyst build? It could always be conditioned on a flag that someone could override if they needed to.
3. Same goes for debug mode - can we implicitly merge the debug plist item, add a condition to not include them if it is explicitly set.

**Comment by rolfbjarne** [comment]
You can add custom entitlements without a plist using the CustomEntitlements item group.

So this:

https://github.com/dotnet/maui/blob/main/src/Templates/src/templates/maui-mobile/Platforms/MacCatalyst/Entitlements.Debug.plist

could be replaced with:

```xml
<ItemGroup Condition="$(TargetFramework.Contains('-maccatalyst')) and '$(Configuration)' == 'Debug'">
    <CustomEntitlements Include="com.apple.security.get-task-allow" Type="boolean" Value="true" />
</ItemGroup>
```

> 2\. Is there a good reason _not_ to implicitly merge the release plist items into a release maccatalyst build?

If the app is published through other means than the App Store, then those entitlements aren't needed (and maybe even just plain wrong).

One idea could be to introduce a helper property to indicate the app developer's intentions:

```xml
<PropertyGroup>
    <AppPublishingIntention>AppStore</AppPublishingIntention> <!-- we set defaults to make release work for the app store -->
    <AppPublishingIntention>AdHoc</AppPublishingIntention> <!-- we set defaults to make release work for adhoc publishing -->
    <!-- if not set, we'll guess something (which may or may not change in the future) -->
</PropertyGroup>
```

There's also the question of identifying a debug or a release or a publish build: it's been a point of friction for customers that we pivot on the Configuration, but nobody has come up with a better solution (https://github.com/dotnet/sdk/issues/31918).

**Comment by aritchie** [comment]
@rolfbjarne Is this CustomEntitlements node documented anywhere?  How do I set multiple values like the StringArray?

**Comment by rolfbjarne** [comment]
@aritchie no, it's not documented yet, but you can set a string array like this:

```xml
<ItemGroup>
    <CustomEntitlements Include="com.apple.security.get-task-allow" Type="stringarray" Value="a;b;c" />
</ItemGroup>
```

More details can be found here: https://github.com/xamarin/xamarin-macios/blob/112318fec4b58f2768b921248767b3920b04484f/msbuild/Xamarin.MacDev.Tasks/Tasks/CompileEntitlementsTaskBase.cs#L256-L263

**Comment by aritchie** [comment]
@rolfbjarne Much appreciated!  This works better than docs anyhow

**Comment by zippo227** [comment]
This helped me work through an issue where I had to use multiple entitlement files to support push notifications on iOS. Now I'm using the ItemGroup version here, and it works within my Github Action to build the maui app.

**Comment by aritchie** [comment]
@rolfbjarne one last issue I'm seeing - 

```xml
<CustomEntitlements Include="keychain-access-groups" Type="StringArray" Value="$(AppIdentifierPrefix)com.mydomain" />
```

Obviously, $(AppIdentifierPrefix) is replaced to "" by msbuild.  I tried an escape of %24(AppIdentifierPrefix) which did set the entitlement, but the teamID is never set.  Manually entering the teamID here does work obviously, but isn't ideal.  Thoughts?
